### ISSUE TITLE
[Epic] Improve Appsmith navigation

### ISSUE THREAD

**Issue by areyabhishek** [appsmithorg/appsmith#4051]
## Problem statement
1. New users don't know where to click once they sign up for Appsmith and land on the editor.
2. Repeat users struggle to move quickly between entities they are using.
3. Users can't easily return from the datasource to the property pane of the widget they were configuring.
## Success criteria
If first time users can easily find their way to the correct datasource section and find their way back to the widget pane easily.
If repeat users can easily navigate through datasources and widgets even when there are lots of them in the entity explorer.

**Comment by Nikhil-Nandagopal** [comment]
Closing as duplicate of #3299
